src/porepy/numerics/mixed_dim/coupler.py

```diff
@@ -101,13 +112,24 @@ def split(self, gb, key, values):
 
         """
         dofs = np.empty(gb.size(), dtype=int)
-        for _, d in gb:
-            dofs[d['node_number']] = d['dof']
+        for _, data in gb:
+            k_data = self.__get_data(data, data_key)
+            dofs[data['node_number']] = k_data['dof']
         dofs = np.r_[0, np.cumsum(dofs)]
 
-        gb.add_node_prop(key)
+        if data_key is None:
+            gb.add_node_prop(key)
+
         for g, d in gb:
-            i = d['node_number']
-            d[key] = values[slice(dofs[i], dofs[i+1])]
+            k_data = self.__get_data(data, data_key)
+            i = data['node_number']
+            k_data[key] = values[slice(dofs[i], dofs[i+1])]
 
 #------------------------------------------------------------------------------#
+    def __get_data(self, data, data_key):
```

Comment:
Perhaps it makes sense. But what do you do when you need more than one subdict within a solver, e.g. Biot? 

I'm also not convinced to tie this too tightly with the GridBucket, since the transition to a single grid should be seamless.